### PR TITLE
Include Razor editor config for design time builds

### PR DESCRIPTION
**Pull request by captainsafia**
Part of https://github.com/dotnet/aspnetcore/issues/31155.

Includes a `RazorSourceGenerator.razorencconfig` file with `SuppressRazorSourceGenerator=true` property in `DesignTimeBuild`-enabled builds.

### CODE REVIEW

File: src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.SourceGenerators.targets [hunk 1]
```diff
@@ -19,6 +19,12 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_RazorSdkToolsDirectoryRoot>$(RazorSdkDirectoryRoot)\tools\</_RazorSdkToolsDirectoryRoot>
     </PropertyGroup>
 
+    <PropertyGroup>
+      <RazorEncConfigFile>$(_RazorSdkSourceGeneratorDirectoryRoot)RazorSourceGenerator.razorencconfig</RazorEncConfigFile>
+    </PropertyGroup>
+
+    <EditorConfigFiles Include="$(RazorEncConfigFile)" Condition="'$(DesignTimeBuild)' == 'true'"/>
```
Comment (captainsafia) [inline src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.SourceGenerators.targets:26]: @pranavkm I don't think we need to do a building in VS check for this case but let me know if my assumption is incorrect.

Comment (pranavkm) [reply src/RazorSdk/Targets/Microsoft.NET.Sdk.Razor.SourceGenerators.targets:26]: It might not hurt to toss that in for good measure.

**Review by pranavkm — COMMENTED** [review]
This PR needs an additional change in the source generator to no-op if the `SuppressRazorSourceGenerator ` is configured

**Review by pranavkm — APPROVED** [review]
Looks good to me. 

@tmat / @jasonmalinowski could you also sign off on these changes?

**Review by tmat — APPROVED** [review]
:shipit:

File: src/RazorSdk/SourceGenerators/RazorSourceGenerationContext.cs [hunk 2]
```diff
@@ -61,6 +71,8 @@ public RazorSourceGenerationContext(GeneratorExecutionContext context)
 
             globalOptions.TryGetValue("build_property._RazorSourceGeneratorDebug", out var waitForDebugger);
 
+            globalOptions.TryGetValue("build_property.SuppressRazorSourceGenerator", out var suppressRazorSourceGenerator);
```
Comment (jasonmalinowski) [inline src/RazorSdk/SourceGenerators/RazorSourceGenerationContext.cs:74]: You don't strictly need the build_property prefix here -- that's the prefix used for MSBuild properties that are mirrored in. So if you're really worried about something trying to do that and breaking something, that might be bad. (It also might be helpful maybe if you're trying to work around something or for testing.)

**Review by jasonmalinowski — COMMENTED** [review]
This looks great!

**Pull request merged**